protocols/imap/src/main/java/org/apache/james/imap/processor/GetAnnotationProcessor.java

```diff
@@ -0,0 +1,139 @@
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.imap.processor;
+
+import com.google.common.base.Function;
+import com.google.common.base.Predicate;
+import com.google.common.collect.FluentIterable;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSortedSet;
+import org.apache.james.imap.api.ImapCommand;
+import org.apache.james.imap.api.ImapConstants;
+import org.apache.james.imap.api.ImapSessionUtils;
+import org.apache.james.imap.api.display.HumanReadableText;
+import org.apache.james.imap.api.message.response.StatusResponse;
+import org.apache.james.imap.api.message.response.StatusResponseFactory;
+import org.apache.james.imap.api.process.ImapProcessor;
+import org.apache.james.imap.api.process.ImapSession;
+import org.apache.james.imap.message.request.GetAnnotationRequest;
+import org.apache.james.imap.message.response.AnnotationResponse;
+import org.apache.james.mailbox.MailboxManager;
+import org.apache.james.mailbox.exception.MailboxException;
+import org.apache.james.mailbox.exception.MailboxNotFoundException;
+import org.apache.james.mailbox.model.MailboxAnnotation;
+import org.apache.james.mailbox.model.MailboxPath;
+import org.slf4j.Logger;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Set;
+
+public class GetAnnotationProcessor extends AbstractMailboxProcessor<GetAnnotationRequest> implements CapabilityImplementingProcessor {
+    public GetAnnotationProcessor(ImapProcessor next, MailboxManager mailboxManager, StatusResponseFactory factory) {
+        super(GetAnnotationRequest.class, next, mailboxManager, factory);
+    }
+
+    public List<String> getImplementedCapabilities(ImapSession session) {
+        return ImmutableList.of(ImapConstants.SUPPORTS_ANNOTATION);
+    }
+
+    protected void doProcess(GetAnnotationRequest message, ImapSession session, String tag, ImapCommand command,
+            Responder responder) {
+        final String mailboxName = message.getMailboxName();
+        try {
+            MailboxPath mailboxPath = buildFullPath(session, mailboxName);
+            List<MailboxAnnotation> mailboxAnnotations = getMailboxAnnotations(session, message.getKeys(), message.getDepthValue(), mailboxPath);
+
+            final long maxsize = message.getMaxsizeValue();
+            if (maxsize > 0) {
+                ImmutableSortedSet<Integer> maxsizeValues = getValuesOverSize(mailboxAnnotations, maxsize);
```

Comment:
No need to be so precise with the type here, it can just be typed as Set I guess